src/robot_interface/robot_interface.py

```diff
@@ -1,146 +1,214 @@
+    def schedule_step(self, step: Step) -> None:
+        """Schedules a step on the robot.
+
+        The method must adapt the standard step to the
+        specific robots mission planning.
 
-        If a mission is scheduled it may be that another mission
-        should not be scheduled.
+        Parameters
+        ----------
+        step : Step
+            Step object that describes the intended task for the
+            robot to perform.
 
         Returns
         -------
-        bool
-            True if a mission is already scheduled, false otherwise.
+        None
+
+        Raises
+        ------
+        RobotException if function fails
         """
         raise NotImplementedError
 
     @abstractmethod
-    def mission_status(self, mission_id: Any) -> MissionStatus:
-        """Retrieves status of the current executing mission for the robot.
+    def step_status(self, step: Optional[Step] = None) -> StepStatus:
+        """Function that returns the status of the progress for
+        a given step.
 
         Parameters
         ----------
-        mission_id : Any
-            Unique identifier for the mission which the status should be checked for
+        step : Optional[Step]
+            Step object that indicate which step to check status for.
+            None indicates current step.
 
         Returns
         -------
-        MissionStatus
-            Enum member indicating the current mission status.
+        StepStatus
+            The status of the desired step
+
+        Raises
+        ------
+        RobotException if function fails
         """
         raise NotImplementedError
 
     @abstractmethod
-    def abort_mission(self) -> bool:
-        """Aborts the current mission for the robot.
+    def get_inspection_step_results(
+        self, inspection_step: InspectionStep
+    ) -> Sequence[InspectionResult]:
+        """Get the inspection result data from a scheduled inspection
+        step.
 
-        If another mission is queued, it should not start before a verification
-        is sent.
+        Parameters
+        ----------
+        inspection_step : InspectionStep
+            The Step that we want to get the inpection results for.
 
         Returns
         -------
-        bool
-            True on successful abort of mission, false otherwise.
+        Sequence[InspectionResult]
+            A list containing all the insepction results connected to
+            a the provided step.
+
+        Raises
+        ------
+        RobotException if function fails
         """
         raise NotImplementedError
 
     @abstractmethod
-    def log_status(
-        self, mission_id: Any, mission_status: MissionStatus, current_step: Step
-    ) -> None:
-        """Logs the status of the executing mission that is being monitored.
+    def get_sensor_recordings(self, sensor: Sensor) -> Sequence[InspectionResult]:
+        """Get the recorded data from the given sensor.
```

Comment:
Why do we need both of these methods? When will we retrieve results that are not due to an inspection step but we need to link directly to a sensor?